orttraining/orttraining/python/experimental/checkpoint.py

```diff
@@ -0,0 +1,280 @@
+from collections import OrderedDict
+import numpy as np
+import onnx
+import os
+import torch
+import warnings
+
+
+################################################################################
+# Experimental Checkpoint APIs
+################################################################################
+
+
+def experimental_state_dict(ort_trainer):
+    if not ort_trainer._training_session:
+        warnings.warn("ONNX Runtime training session is not initialized yet. "
+                        "Please run train_step or eval_step at least once before calling state_dict().")
+        return {}
+
+    # extract trained weights
+    session_state = ort_trainer._training_session.get_state()
+    torch_state = {}
+    for name in session_state:
+        torch_state[name] = torch.from_numpy(session_state[name])
+
+    # extract untrained weights and buffer
+    for n in ort_trainer._onnx_model.graph.initializer:
+        if n.name not in torch_state:
+            torch_state[n.name] = torch.from_numpy(np.array(onnx.numpy_helper.to_array(n)))
+
+    # Need to remove redundant initializers and name suffices to map back to original torch state names
+    torch_state_to_return = {key: torch_state[key] for key in ort_trainer._original_model_state_keys if key in torch_state} \
+                            if ort_trainer._original_model_state_keys else torch_state
+    return torch_state_to_return
+
+def experimental_load_state_dict(ort_trainer, state_dict, strict=False):
```

Comment:
Hi @mrry - is this naming convention aligned with what you had in mind?

Reply:
This works for me! (Originally I had imagined that we'd just have `onnxruntime.experimental.load_state_dict()` etc. without the `experimental_` on the function name, but I actually prefer this "uglier"/more verbose version, since it'll be easier to grep for and more robust to namespace aliasing.)